agent-lib/src/main/java/org/terracotta/angela/agent/AgentController.java

```diff
@@ -725,4 +717,45 @@ private void zipFolder(ZipOutputStream zos, String parent, File folder) throws I
   public Map<String, ?> getNodeAttributes() {
     return ignite.configuration().getUserAttributes();
   }
+
+  private static class Dirs {
+    final File kitDir;
+    final File workingDir;
+
+    public Dirs(File kitDir, File workingDir) {
+      this.kitDir = kitDir;
+      this.workingDir = workingDir;
+    }
+
+    static Optional<Dirs> discover(InstanceId instanceId, String hostName, Distribution distribution, License license, String kitInstallationName, String kitInstallationPath) {
+      if (kitInstallationPath == null) {
+        RemoteKitManager kitManager = new RemoteKitManager(instanceId, distribution, kitInstallationName);
+        if (!kitManager.isKitAvailable()) {
+          return Optional.empty();
+        }
+
+        logger.info("Installing kit on host {} from {}", hostName, distribution);
+        return Optional.of(new Dirs(
+            kitManager.installKit(license, Collections.singletonList(hostName)),
+            kitManager.getWorkingDir().toFile()
+        ));
+
+      } else {
+        File kitDir = new File(kitInstallationPath);
+        if (license != null) {
+          license.writeToFile(kitDir);
```

Comment:
@chrisdennis : quick note: Angela was "updating" the kit into `kits/<version>/tc-db-kit` to add the license file if given, usually taken from `src/test/resources`.
When running concurrent tests, this might create issues... A file check is done first. So it could be better to (when preparing the test kit) have Gradle copy this license file from the test resources into the exploded kit.